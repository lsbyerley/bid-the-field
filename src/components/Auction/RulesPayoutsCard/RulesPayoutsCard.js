import { useAppContext } from '../../../AppContext';
import useAsyncReference from '../../../lib/useAsyncReference';

const RulesPayoutsCard = ({ auction }) => {
  const asyncAuction = useAsyncReference(auction, true);
  const { setModalOpen, setModalContent } = useAppContext();

  const openRulesModal = () => {
    setModalContent(asyncAuction.current.rules);
    setModalOpen(true);
  };

  const openPayoutsModal = () => {
    setModalContent(asyncAuction.current.payouts);
    setModalOpen(true);
  };
  return (
    <div className='rounded-lg card compact bg-base-200 '>
      <div className='justify-center card-body'>
        {asyncAuction.current.rules && (
          <button
            className='btn btn-ghost btn-sm'
            onClick={() => openRulesModal()}
          >
            Rules
          </button>
        )}
        {asyncAuction.current.payouts && (
          <button
            className='btn btn-ghost btn-sm'
            onClick={() => openPayoutsModal()}
          >
            Payouts
          </button>
        )}
      </div>
    </div>
  );
};

export default RulesPayoutsCard;
