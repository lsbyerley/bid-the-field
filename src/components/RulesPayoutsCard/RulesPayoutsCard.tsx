import { useAppContext } from '@/AppContext';
import useAsyncReference from '@/lib/useAsyncReference';

import type { RulesPayoutsCardArgs } from '@/types';

const RulesPayoutsCard = ({ auction }: RulesPayoutsCardArgs) => {
  const asyncAuction = useAsyncReference(auction, true);
  const { setModalOpen, setModalContent } = useAppContext();

  const openRulesModal = () => {
    setModalContent(asyncAuction.current.rules);
    setModalOpen(true);
  };

  const openPayoutsModal = () => {
    setModalContent(asyncAuction.current.payouts);
    setModalOpen(true);
  };
  return (
    <div className='rounded-lg card compact bg-base-100 '>
      <div className='justify-center card-body'>
        {asyncAuction.current.rules && (
          <button
            className='btn btn-ghost btn-sm no-animation'
            onClick={() => openRulesModal()}
          >
            Rules
          </button>
        )}
        {asyncAuction.current.payouts && (
          <button
            className='btn btn-ghost btn-sm no-animation'
            onClick={() => openPayoutsModal()}
          >
            Payouts
          </button>
        )}
      </div>
    </div>
  );
};

export default RulesPayoutsCard;
