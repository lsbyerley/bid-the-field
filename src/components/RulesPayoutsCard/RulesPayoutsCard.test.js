import React from 'react';
import { render, screen } from 'test-utils';
import RulesPayoutsCard from './RulesPayoutsCard';

const defaultProps = (overrides) => {
  return {
    auction: {
      rules: 'test',
      payouts: 'test',
    },
    ...overrides,
  };
};

const renderRulesPayoutsCard = (props) => {
  const utils = render(<RulesPayoutsCard {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('RulesPayoutsCard', () => {
  it('should render and match snapshot', () => {
    const { home } = renderRulesPayoutsCard(defaultProps());
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
