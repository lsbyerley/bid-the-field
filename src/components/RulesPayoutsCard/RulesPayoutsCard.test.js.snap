// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RulesPayoutsCard should render and match snapshot 1`] = `
<div
  class="rounded-lg card compact bg-base-100 "
>
  <div
    class="justify-center card-body"
  >
    <button
      class="btn btn-ghost btn-sm"
    >
      Rules
    </button>
    <button
      class="btn btn-ghost btn-sm"
    >
      Payouts
    </button>
  </div>
</div>
`;
